src/Profiles.ts

```diff
@@ -336,6 +338,178 @@ export class Profiles {
+        };
+        // confirm that the user really wants to delete
+        if (await vscode.window.showQuickPick([localize("deleteProfile.showQuickPick.yes", "Yes"),
+            localize("deleteProfile.showQuickPick.no", "No")], quickPickOptions) !== localize("deleteProfile.showQuickPick.yes", "Yes")) {
+            this.log.debug(localize("deleteProfile.showQuickPick.log.debug", "User picked no. Cancelling delete of profile"));
+            return;
+        }
+
+        const profileType = ZoweExplorerApiRegister.getMvsApi(deletedProfile).getProfileTypeName();
+        try {
+            this.deleteProf(deletedProfile, profileName, profileType);
+        } catch (error) {
+            this.log.error(localize("deleteProfile.delete.log.error", "Error encountered when deleting profile! ") + JSON.stringify(error));
+            await errorHandling(error, profileName, error.message);
+            throw error;
+        }
+
+        vscode.window.showInformationMessage("Profile " + profileName + " was deleted.");
+        return profileName;
+    }
+
+    public async deleteProf(ProfileInfo, ProfileName, ProfileType) {
+        let zosmfProfile: IProfile;
+        try {
+            zosmfProfile = await (await this.getCliProfileManager(ProfileType))
+            .delete({ profile: ProfileInfo, name: ProfileName, type: ProfileType });
+        } catch (error) {
+            vscode.window.showErrorMessage(error.message);
+        }
+        return zosmfProfile.profile;
+    }
+
+    public async deleteProfile(
+        datasetTree: IZoweTree<IZoweDatasetTreeNode>, ussTree: IZoweTree<IZoweUSSTreeNode>,
+        jobsProvider: IZoweTree<IZoweJobTreeNode>, node?: IZoweNodeType) {
+
+            let deleteLabel: string;
+            let deletedProfile: IProfileLoaded;
+            if (!node){
+                deletedProfile = await this.getDeleteProfile();
+                if (!deletedProfile) {
+                    return;
+                } else {
+                    deleteLabel = deletedProfile.name;
+                }
+            } else {
+                deletedProfile = node.getProfile();
+                deleteLabel = node.getProfileName();
+            }
+            const deleteSuccess = await this.deletePrompt(deletedProfile);
+            if (!deleteSuccess){
+                vscode.window.showInformationMessage(localize("deleteProfile.noSelected",
+                    "Operation Cancelled"));
+                return;
+            }
+
+            // Delete from Data Set Tree
+            datasetTree.mSessionNodes.forEach((sessNode) => {
+                if (sessNode.getProfileName() === deleteLabel) {
+                    datasetTree.deleteSession(sessNode);
+                }
+            });
+            await datasetTree.refresh();
+
+            // Delete from Data Set Favorites
+            datasetTree.mFavorites.forEach((ses) => {
+                const findNode = ses.label.substring(1, ses.label.indexOf("]")).trim();
```

Comment:
This looks like trouble going forward 😁 But I do not know how to do it better at this point.